src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -5,48 +5,60 @@
 
 package net.minecraftforge.common.capabilities;
 
-import net.minecraftforge.fml.ModLoader;
-import net.minecraftforge.forgespi.language.ModFileScanData;
+import net.minecraft.resources.ResourceLocation;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.IdentityHashMap;
-import java.util.List;
 
 import static net.minecraftforge.fml.Logging.CAPABILITIES;
 
+/**
+ * The Capability Manager is responsible for making and 
+ * providing {@link CapabilityType} instances.
+ */
 public enum CapabilityManager
 {
     INSTANCE;
-    static final Logger LOGGER = LogManager.getLogger();
 
+	private static final Logger LOGGER = LogManager.getLogger();
+    private static final IdentityHashMap<ResourceLocation, CapabilityType<?>> PROVIDERS = new IdentityHashMap<>();	
 
-    public static <T> Capability<T> get(CapabilityToken<T> type)
+    /**
+     * Retrieve a {@link CapabilityType}.  The returned type may not be registered.
+     * @param <T> The supertype of all instances of this capability type.
+     * @param id The ID of this capability type.
+     * @return A {@link CapabilityType} for the provided key.
+     */
+    public static <T> @NotNull CapabilityType<T> get(ResourceLocation id)
     {
-        return INSTANCE.get(type.getType(), false);
+        return INSTANCE.get(id, false);
     }
 
+    /**
+     * Internal API for registering capability types.
+     * @see {@link RegisterCapabilitiesEvent}
+     */
     @SuppressWarnings("unchecked")
-    <T> Capability<T> get(String realName, boolean registering)
+    <T> @NotNull CapabilityType<T> get(ResourceLocation id, boolean registering)
     {
-        Capability<T> cap;
+        CapabilityType<T> cap;
 
-        synchronized (providers)
+        synchronized (PROVIDERS)
         {
-            realName = realName.intern();
-            cap = (Capability<T>)providers.computeIfAbsent(realName, Capability::new);
+            cap = (CapabilityType<T>)PROVIDERS.computeIfAbsent(id, CapabilityType::new);
```

Comment:
This does no checks that the IDs are unique? Meaning this just happily allows you to registry replace capabilities?

Reply:
It does not need to - capabilities are not unique, nor can they be replaced.  it is design this way such that a mod can have a capability api where one mod does
cap = manager.get("mymod:cap")
and register it, and another mod can also have
otherModCap = manager.get("mymod:cap")

They can then query the `otherModCap` to see if the cap has been registered, instead of requiring field access to wherever the other mod is storing `cap`.

Reply:
My bad, I was thinking too much about the `IdentityHashMap` and forgot to leave a comment on that explicitly.